fmriprep/utils/tests/test_derivative_cache.py

```diff
@@ -0,0 +1,41 @@
+import json
+from pathlib import Path
+
+import pytest
+
+from fmriprep.data import load as load_data
+from fmriprep.utils import bids
+
+
+@pytest.mark.parametrize('xfm', ['boldref2fmap', 'boldref2anat', 'hmc'])
+def test_transforms_found_as_str(tmp_path: Path, xfm: str):
+    spec = (
+        json.loads(load_data.readable('io_spec.json').read_text())
+        .get('queries')
+        .get('transforms')
+        .get(xfm)
+    )
+    entities = {
+        'subject': '0',
+        'task': 'rest',
+        'suffix': 'bold',
+        'extension': '.nii.gz',
+    }
+    if xfm == 'boldref2fmap':
+        to_find = f'sub-{entities["subject"]}_task-{entities["task"]}_from-{spec["from"]}_to-auto00000_mode-image_xfm.txt'  # noqa: E501
+    else:
+        to_find = f'sub-{entities["subject"]}_task-{entities["task"]}_from-{spec["from"]}_to-{spec["to"]}_mode-image_xfm.txt'  # noqa: E501
+
+    funcd = tmp_path / f'sub-{entities["subject"]}' / 'func'
+    funcd.mkdir(parents=True)
+    (funcd / to_find).touch()
+
+    derivs = bids.collect_derivatives(
+        derivatives_dir=tmp_path,
+        entities=entities,
+        fieldmap_id='auto_00000',
+    )
+    transforms_in_derivs = 'transforms' in derivs
+    xfm_in_transforms = xfm in derivs.get('transforms')
+    transform_is_str = isinstance(derivs.get('transforms').get(xfm), str)
+    assert all((transforms_in_derivs, xfm_in_transforms, transform_is_str))
```

Comment:
You're writing this defensively, as you might production code, but tests generally give more informative output with one assertion per line, and testing what you actually care about. I'd suggest:

```suggestion
    assert isinstance(derivs.get('transforms'), dict)
    assert xfm in derivs['transforms']
    assert derivs['transforms'][xfm] == str(to_find)
```

Reply:
I had been under the impression that aggregating tests with an `all` was generally preferred -- the idea being that the whole test will fail at the first individual failed assert statements, hiding whether later assert statements might have failed, whereas the `all([test0,test1,test2,...])` construct reports the results of all assertions each time.

Reply:
If they can fail independently, then it might be worth separating, typically by parameterizing the test. In this case, you're checking the shape of the response, where one failure would lead to the next, so there's no advantage to checking them independently.

Which makes me realize, we could also just write:

```suggestion
    assert derivs == {'transforms': {xfm: str(to_find)}}
```